thunder/transforms/quantization.py

```diff
@@ -294,3 +295,214 @@ def transform_traces_pre_prologue(self, prologue_trace, computation_trace, epilo
+
+        new_computation_trace.args = (*new_computation_trace.args, *new_compute_inputs)
+        new_computation_trace._siginfo.args = [(a.name, None) for a in new_computation_trace.args]
+
+        with tracectx(new_computation_trace):
+            new_bindings = [
+                thunder.core.prims.unpack_trivial.bind(i, output=i, name=i.name) for i in new_compute_inputs
+            ]
+
+        for idx, bsym in enumerate(bound_symbols):
+            if bsym.sym != prims.unpack_trivial:
+                break
+            new_computation_trace.bound_symbols.append(bsym.from_bsym())
+        new_computation_trace.bound_symbols += new_bindings
+
+        for bsym in bound_symbols[idx:]:
+            if bsym.sym == thunder.torch.linear:
+                n = lora_linear_proxies[bsym.args[1].name]
+                with thunder.core.trace.tracectx(computation_trace):
+                    loraA_transpose_meta = prims.transpose.meta(additional_proxies[f"{n}.loraA"], (1, 0))
+                    loraA_bsym = prims.transpose.bind(
+                        additional_proxies[f"{n}.loraA"],
+                        (1, 0),
+                        output=(loraA_transpose_meta),
+                    )
+                    loraB_transpose_meta = prims.transpose.meta(additional_proxies[f"{n}.loraB"], (1, 0))
+                    loraB_bsym = prims.transpose.bind(
+                        additional_proxies[f"{n}.loraB"],
+                        (1, 0),
+                        output=(loraB_transpose_meta),
+                    )
+                    lora_meta_ = prims.matmul.meta(bsym.args[0], loraA_bsym.output)
+                    lora_bsym_ = prims.matmul.bind(bsym.args[0], loraA_bsym.output, output=lora_meta_)
+                    lora_meta = prims.matmul.meta(lora_bsym_.output, loraB_bsym.output)
+                    lora_bsym = prims.matmul.bind(lora_bsym_.output, loraB_bsym.output, output=lora_meta)
+
+                    # Is there a better way?
+                    original_weight_meta = prims.linear.meta(*bsym.args[:3])
+                    original_weight = prims.linear.bind(*bsym.args[:3], output=original_weight_meta)
+
+                    lora_sum_bsym = bsym.from_bsym(
+                        sym=prims.add,
+                        args=(original_weight.output, lora_bsym.output),
+                    )
+
+                new_computation_trace.bound_symbols.append(loraA_bsym)
+                new_computation_trace.bound_symbols.append(loraB_bsym)
+                new_computation_trace.bound_symbols.append(lora_bsym_)
+                new_computation_trace.bound_symbols.append(lora_bsym)
+                new_computation_trace.bound_symbols.append(original_weight)
+                new_computation_trace.bound_symbols.append(lora_sum_bsym)
+            else:
+                new_computation_trace.bound_symbols.append(bsym.from_bsym())
+
+        new_computation_trace.set_provenance(thunder.core.trace.TraceProvenance("lora linear pass"))
```

Comment:
Maybe we could also set the provenance for the (copied new) prologue trace.

Reply:
Ok! Let me add that to the new prologue trace!